be/src/exec/pipeline/pipeline_driver_executor.cpp

```diff
@@ -209,6 +219,33 @@ void GlobalDriverExecutor::_worker_thread() {
     }
 }
 
+StatusOr<DriverRawPtr> GlobalDriverExecutor::_get_next_driver(std::queue<DriverRawPtr>& local_mutable_driver_queue) {
+    DriverRawPtr driver = nullptr;
+    if (!local_mutable_driver_queue.empty()) {
+        const size_t mutable_driver_num = local_mutable_driver_queue.size();
+        for (size_t i = 0; i < mutable_driver_num; i++) {
+            driver = local_mutable_driver_queue.front();
+            local_mutable_driver_queue.pop();
+            if (driver->source_operator()->has_output()) {
+                return driver;
+            } else {
+                if (driver->driver_acct().get_mutable_queue_time_spent() > MUTABLE_MAX_WAIT_TIME_SPENT) {
+                    driver->driver_acct().clean_mutable_infos();
+                    driver->set_driver_state(DriverState::INPUT_EMPTY);
+                    _blocked_driver_poller->add_blocked_driver(driver);
+                } else {
+                    local_mutable_driver_queue.push(driver);
+                }
+                driver = nullptr;
+            }
+        }
+    }
+
+    // If mutable driver queue is not empty, we cannot block here.
```

Comment:
why?

Reply:
Otherwise these drivers may not be scheduled until the ready queue is not empty.